pac4j-oauth/src/main/java/org/pac4j/oauth/client/QQClient.java

```diff
@@ -0,0 +1,106 @@
+package org.pac4j.oauth.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.pac4j.oauth.profile.qq.QQProfile;
+import org.pac4j.oauth.profile.qq.QQProfileCreator;
+import org.pac4j.oauth.profile.qq.QQProfileDefinition;
+import org.pac4j.scribe.builder.api.QQApi20;
+
+/**
+ * <p>This class is the OAuth client to authenticate users in Tencent QQ Connect.</p>
+ * <p>It returns a {@link QQProfile}.</p>
+ * <p>More information at http://wiki.connect.qq.com/%E4%BD%BF%E7%94%A8authorization_code%E8%8E%B7%E5%8F%96access_token</p>
+ *
+ * @author Zhang Zhenli
+ * @since 3.1.0
+ */
+public class QQClient extends OAuth20Client<QQProfile> {
+
+    public enum TencentQQScope {
+        /**
+         * Get the nickname, avatar, and gender of the logged in user
+         */
+        get_user_info,
+        /**
+         * Get basic information about QQ VIP
+         */
+        get_vip_info,
+        /**
+         * Get advanced information about QQ VIP
+         */
+        get_vip_rich_info,
+        /**
+         * Get user QQZone album list
+         */
+        list_album,
+        /**
+         * Upload a photo to the QQZone album
+         */
+        upload_pic,
+        /**
+         * Create a new personal album in the user's QQZone album
+         */
+        add_album,
+        /**
+         * Get a list of photos in the user's QQZone album
+         */
+        list_photo,
+        /**
+         * Get the delivery address of Tenpay users
+         */
+        get_tenpay_addr
+    }
+
+
+    protected List<TencentQQScope> scopes;
```

Comment:
Can we have an empty list by default? I think it could make the code easier and safer.

Reply:
The official documentation has default value “get_user_info”, but your suggestion is necessary.

![image](https://user-images.githubusercontent.com/5005199/42313826-3a8861dc-8076-11e8-9c54-864c7f6b4784.png)